pageserver/src/walredo.rs

```diff
@@ -152,30 +160,46 @@ impl PostgresRedoManager {
             tenant_shard_id,
             conf,
             last_redo_at: std::sync::Mutex::default(),
-            redo_process: RwLock::new(None),
+            redo_process: tokio::sync::RwLock::new(Poison::new("redo_process field", None)),
         }
     }
 
     /// This type doesn't have its own background task to check for idleness: we
     /// rely on our owner calling this function periodically in its own housekeeping
     /// loops.
-    pub(crate) fn maybe_quiesce(&self, idle_timeout: Duration) {
-        if let Ok(g) = self.last_redo_at.try_lock() {
+    pub(crate) async fn maybe_quiesce(&self, idle_timeout: Duration) {
+        // awkward control flow because rustc isn't smart enough to detect that we don't
+        // hold the std::sync lock guard across the await point
```

Comment:
std::sync mention is probably out of date now

Reply:
No, that was referring to the last_redo_at lock guard, it was accurate (I added it in this PR).

The switch to heavier_once_cell makes it unnecessary.